mattermost/mattermost-webapp
# MM-30079 - Hide or Enable System Console Settings

#### Summary

Currently there are five related settings in the System Console that CWS admins cannot change:
- Authentication ➜ Email ➜ Require Email Verification
- Authentication ➜ Signup ➜ Enable Email Invitations
- Site Configurations ➜ Notifications ➜ Enable Email Notifications
- Site Configuration ➜ Users and Teams ➜ Allow users to view archived channels
- Integrations ➜ Integration Management ➜ Enable OAuth 2.0 Service Provider
- Site Configuration → Notifications → Enable Preview Mode Banner
- Site Configuration → Notifications → Enable Email Batching

In all cases the setting is locked as "True" and displays the following text:
- `This setting has been set through an environment variable. It cannot be changed through the System Console.`

#### Test Plan
1. Setup a Mattermost Cloud Server with these changes
2. Navigate to each menu and ensure that the section of the menu described in the above summary no longer appear

#### Ticket Link
[MM-30079](https://mattermost.atlassian.net/browse/MM-30079)

#### Screenshots
|  before  |  after  |
|----|----|
| <img width="1202" alt="Screen Shot 2022-11-01 at 4 44 31 PM" src="https://user-images.githubusercontent.com/116016004/199338608-ab093019-d47c-4037-8fba-ce4a4963ba35.png"> | <img width="1202" alt="Screen Shot 2022-11-01 at 4 49 28 PM" src="https://user-images.githubusercontent.com/116016004/199339052-3b98afb1-eb8c-4c40-8cd6-50bb615c9493.png"> |
| <img width="1202" alt="Screen Shot 2022-11-01 at 4 45 26 PM" src="https://user-images.githubusercontent.com/116016004/199338678-ee8997b0-1712-40c2-be86-589ceef06396.png"> | <img width="1202" alt="Screen Shot 2022-11-01 at 4 49 58 PM" src="https://user-images.githubusercontent.com/116016004/199339131-20fc90f0-8c15-40d2-873d-315c90a6f9a5.png"> |
| <img width="1202" alt="Screen Shot 2022-11-01 at 4 46 42 PM" src="https://user-images.githubusercontent.com/116016004/199338757-e546b4be-d275-4356-b162-eeb981bcd5e3.png"> | <img width="1202" alt="Screen Shot 2022-11-01 at 4 50 25 PM" src="https://user-images.githubusercontent.com/116016004/199339197-e51c0a8e-99d1-407f-aed3-b3341c300e1b.png"> |
| <img width="1202" alt="Screen Shot 2022-11-01 at 4 47 31 PM" src="https://user-images.githubusercontent.com/116016004/199338816-977db07c-ce40-4532-b6e1-5d5a7009d25b.png"> | <img width="1202" alt="Screen Shot 2022-11-01 at 4 51 37 PM" src="https://user-images.githubusercontent.com/116016004/199339505-985c2189-fd76-4f4d-b2dc-58545291be3d.png"> |
| <img width="1202" alt="Screen Shot 2022-11-01 at 4 48 32 PM" src="https://user-images.githubusercontent.com/116016004/199338872-f342f968-af4d-44c5-a4fb-fef78e5d7609.png"> | <img width="1202" alt="Screen Shot 2022-11-01 at 4 51 17 PM" src="https://user-images.githubusercontent.com/116016004/199339539-6c69700f-74e3-492d-bfad-f55662272203.png"> | 

#### Release Note

```release-note
NONE
```
